libspecr/src/write/mod.rs

```diff
@@ -0,0 +1,50 @@
+use std::{io::{Write, Stdout, Stderr, Error}, cell::RefCell, fmt};
+
+use crate::*;
+
+/// An object that fulfills both GcCompat and Write.
+pub trait GcWrite: GcCompat + Write {}
+impl<T> GcWrite for T where T: GcCompat + Write {}
+
+#[derive(Copy, Clone, GcCompat)]
+/// Garbage-collected datastructure representing a write stream and implementing `Copy`.
```

Comment:
```suggestion
/// Garbage-collected data structure representing a write stream and implementing `Copy`.
#[derive(Copy, Clone, GcCompat)]
```
Doc comments usually go before attributes